src/handlers.rs

```diff
@@ -701,3 +702,20 @@ fn handle_symlink(i: *const isolate, base: &msg::Base) -> Box<Op> {
     }()))
   }
 }
+
+fn handle_truncate(i: *const isolate, base: &msg::Base) -> Box<Op> {
+  let deno = from_c(i);
+  if !deno.flags.allow_write {
+    return odd_future(permission_denied());
+  }
+  let msg = base.msg_as_truncate().unwrap();
+  let name = msg.name().unwrap();
+  let len = msg.len();
+  Box::new(futures::future::result(|| -> OpResult {
+    debug!("handle_truncate {} {}", name, len);
+    let mut options = fs::OpenOptions::new();
```

Comment:
What about using this https://doc.rust-lang.org/std/fs/struct.OpenOptions.html#method.truncate

Reply:
I believe that method would truncate the whole file to length 0